build-conventions-secant/src/main/kotlin/secant.publish-conventions.gradle.kts

```diff
@@ -0,0 +1,395 @@
+                        .setVersionCodes(bundleVersionCodes)
+                        .setStatus(status)
+                    )
+                )
+            )
+
+        val updatedTrack: Track = updateTrackRequest.execute()
+        log("Track ${updatedTrack.track} has been updated")
+
+        // Commit changes for edit
+        val commitRequest: AndroidPublisher.Edits.Commit = edits.commit(
+            packageName,
+            editId
+        )
+        val appEdit: AppEdit = commitRequest.execute()
+        log("App edit with id ${appEdit.id} has been committed")
+    }
+
+    @TaskAction
+    fun runTask() {
+        log("Publish starting for track: $track and status: $status")
+        runPublish(
+            track,
+            status,
+            gpServiceAccountKey,
+            gpPublisherApiKey
+        )
+        log("Publishing done")
+    }
+}
+
+/**
+ * The release track identifier. This class also serves as a type-safe custom task input validation.
+ */
+enum class PublishTrack {
+    INTERNAL,   // Internal testing track
+    ALPHA,  // Closed testing track
+    BETA,   // Opened testing track. Note that use of this track is not supported by this task.
+    PRODUCTION; // Production track. Note that use of this track is not supported by this task.
+
+    companion object {
+        @Throws(IllegalArgumentException::class)
+        fun new(identifier: String): PublishTrack {
+            // Throws IllegalArgumentException if the specified name does not match any of the defined enum constants
+            return values().find { it.name.lowercase() == identifier }
+                ?: throw IllegalArgumentException("Unsupported enum value: $identifier")
+        }
+    }
+
+    @Throws(IllegalStateException::class)
+    fun toGooglePlayIdentifier(): String {
+        return when (this) {
+            INTERNAL -> "internal"  // $NON-NLS-1$
+            ALPHA -> "alpha"    // $NON-NLS-1$
+            BETA -> error("For security reasons, this script does not support the $this option. Promote the app " +
+                "manually from a lower testing channel instead.")
+            PRODUCTION -> error("For security reasons, this script does not support the $this option. Promote the app" +
+                " manually from a lower testing channel instead.")
+        }
+    }
+}
+
+/**
+ * The status of a release. This class also serves as a type-safe custom task input validation.
+ */
+enum class PublishStatus {
+    STATUS_UNSPECIFIED, // Unspecified status.
+    DRAFT,  // The release's APKs are not being served to users.
+    IN_PROGRESS,    // The release's APKs are being served to a fraction of users, determined by 'userFraction'.
+    HALTED, // The release's APKs will no longer be served to users. Users who already have these APKs are unaffected.
+    COMPLETED;  // The release will have no further changes. Its APKs are being served to all users, unless they are
+    // eligible to APKs of a more recent release.
```

Comment:
```suggestion
    DRAFT,              // The release's APKs are not being served to users.
    IN_PROGRESS,        // The release's APKs are being served to a fraction of users, determined by 'userFraction'.
    HALTED,             // The release's APKs will no longer be served to users. Users who already have these
                        // APKs are unaffected.
    COMPLETED;          // The release will have no further changes. Its APKs are being served to all users,
                        // unless they are eligible for APKs of a more recent release.
```

Reply:
Unfortunately, this is how [Ktlint](https://github.com/pinterest/ktlint) formats and requires comments to be used.